### ISSUE TITLE
REPL: stdlib binPkg symbol clashes with symbol in imported source package

### ISSUE THREAD

**Issue by bailsman** [traefik/yaegi#1098]
The following package `/go/src/sample/sample.go` works fine outside of the REPL:

```go
package main

import (
        "fmt"
        "github.com/ethereum/go-ethereum/common/math"
)

var kaboom math.Decimal256

func main() {
        fmt.Println(kaboom)
}
```

Yaegi gives the expected result:
```console
$ yaegi run sample
{false []}
```

However, it doesn't work in the REPL:

```console
$ yaegi
> import "sample"
1:21: import "sample" error: /go/src/sample/sample.go:8:12: undefined selector math.Decimal256
```

I've managed to work around the problem on my system with the following patch:

```diff
diff --git a/interp/type.go b/interp/type.go
index a2c9b6c..a65d5da 100644
--- a/interp/type.go
+++ b/interp/type.go
@@ -463,17 +464,14 @@ func nodeType(interp *Interpreter, sc *scope, n *node) (*itype, error) {
                }
 
        case identExpr:
-               sym, _, found := sc.lookup(n.ident)
+               // also try with the filename, in case ident is a package name.
+               baseName := filepath.Base(interp.fset.Position(n.pos).Filename)
+               ident := filepath.Join(n.ident, baseName)
+               sym, _, found := sc.lookup(n.ident, ident)
                if !found {
-                       // retry with the filename, in case ident is a package name.
-                       baseName := filepath.Base(interp.fset.Position(n.pos).Filename)
-                       ident := filepath.Join(n.ident, baseName)
-                       sym, _, found = sc.lookup(ident)
-                       if !found {
-                               t.incomplete = true
-                               sc.sym[n.ident] = &symbol{kind: typeSym, typ: t}
-                               break
-                       }
+                       t.incomplete = true
+                       sc.sym[n.ident] = &symbol{kind: typeSym, typ: t}
+                       break
                }
                t = sym.typ
                if t.incomplete && t.cat == aliasT && t.val != nil && t.val.cat != nilT {
diff --git a/interp/scope.go b/interp/scope.go
index bfeeba5..472c6c3 100644
--- a/interp/scope.go
+++ b/interp/scope.go
@@ -126,10 +126,18 @@ func (s *scope) upperLevel() *scope {
 // lookup searches for a symbol in the current scope, and upper ones if not found
 // it returns the symbol, the number of indirections level from the current scope
 // and status (false if no result).
-func (s *scope) lookup(ident string) (*symbol, int, bool) {
+func (s *scope) lookup(ident string, alternativeIdents ...string) (*symbol, int, bool) {
        level := s.level
        for {
-               if sym, ok := s.sym[ident]; ok {
+               sym, ok := s.sym[ident]
+               if !ok {
+                       for _, alternative := range alternativeIdents {
+                               if sym, ok = s.sym[alternative]; ok {
+                                       break
+                               }
+                       }
+               }
+               if ok {
                        if sym.global {
                                return sym, globalFrame, true
                        }
diff --git a/interp/cfg.go b/interp/cfg.go
index 1301375..05e4c6f 100644
--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -1245,14 +1245,10 @@ func (interp *Interpreter) cfg(root *node, importPath string) ([]*node, error) {
                                break
                        }
 
-                       sym, level, found := sc.lookup(n.ident)
+                       sym, level, found := sc.lookup(n.ident, filepath.Join(n.ident, baseName))
                        if !found {
-                               // retry with the filename, in case ident is a package name.
-                               sym, level, found = sc.lookup(filepath.Join(n.ident, baseName))
-                               if !found {
-                                       err = n.cfgErrorf("undefined: %s", n.ident)
-                                       break
-                               }
+                               err = n.cfgErrorf("undefined: %s", n.ident)
+                               break
                        }
                        // Found symbol, populate node info
                        n.typ, n.findex, n.level = sym.typ, sym.index, level
```

**Comment by mvertes** [comment]
I run this test successfully today on master (33a532e), both in batch and using REPL. Can you confirm ?

**Comment by bailsman** [comment]
Confirm fixed by cdc6b773c2024d1cd0e78fa32e3f46723a65674e
